transformers/file_utils.py

```diff
@@ -103,6 +105,18 @@ def docstring_decorator(fn):
             return fn
         return docstring_decorator
 
+
+def is_remote_url(url_or_filename):
+    parsed = urlparse(url_or_filename)
+    return parsed.scheme in ('http', 'https', 's3')
+
+def hf_bucket_url(identifier, postfix=None):
+    if postfix is None:
+        return "/".join((S3_BUCKET_PREFIX, identifier))
+    else:
+        return "/".join((S3_BUCKET_PREFIX, identifier, postfix))
```

Comment:
So, this is the main discussion point, @thomwolf @LysandreJik @mfuntowicz. 

This works with the format where users upload whole directories that are the output of `save_pretrained()`. This seems to me to be a better, future-proof choice than doing for instance `root/{user}/{identifier}-config.json`.

This will, however:
- not work out of the box with the models that are at the root of the bucket currently (the ones we upload). Not a big deal as we have shortcuts for them anyways.
- require me to change the upload CLI so that it uploads a whole folder at once, keeping the folder name.

Reply:
I really like that. I think having the same architecture between `save_pretrained` and the S3 makes sense.

Reply:
I like it too as it gives more visibility to the contributors.